src/ipc/cc_version.c

```diff
@@ -19,18 +19,22 @@
 	field = CC_OPTIMIZE_FLAGS, \
 	field[ARRAY_SIZE(((struct sof_ipc_cc_version *)(0))->optim) - 1] = 0
 
+/* copy CC_DESC to arrays during compilation time */
+#define CC_DESC_COPY(field) \
+	field = CC_DESC, \
+	field[ARRAY_SIZE(((struct sof_ipc_cc_version *)(0))->desc) - 1] = 0
+
 const struct sof_ipc_cc_version cc_version
 	__section(".fw_ready_metadata") = {
 	.ext_hdr = {
 		.hdr.cmd = SOF_IPC_FW_READY,
-		.hdr.size = ALIGN_UP(sizeof(struct sof_ipc_cc_version)
-				     + sizeof(CC_DESC), 4),
```

Comment:
I'd be interested to understand this a bit better. It looks like the respective PR isn't in master yet - what's the definition of `CC_DESC`? If it is indeed something for which `sizeof()` doesn't deliver the desired result, we could just replace that and preserve the flexible size of `.desc` But yes, if we're sure 24 bytes are enough, a fixed size array in the ABI is better.

Reply:
`sizeof(CC_DESC)` is ok, problem is with lack of `\0` after const string in memory dump (`CC_DESC` is defined by build system and after preprocesing is replaced with ` RG-2017.8-linux`)

Reply:
sizeof for string literals should include the \0 actually. I think the intent here is to move to a fixed length which should fit for any compiler.

Let's hope no CC_DESC string has length 24 or more. Then the \0 would disappear from the initializer and length 25 and more might give a warning for initializing a too small array.

Reply:
@paulstelian97 @ktrzcinx I think with current solution, will fill force the last char to be 0 with `CC_DESC_COPY`

Reply:
@lyakh @ktrzcinx @paulstelian97 upate with some more test result in https://github.com/thesofproject/sof/issues/2507
Please help to check.